### PR TITLE
Feat/action config (ability to pass a config object and disable immer) #628

### PR DESCRIPTION
**Pull request by s900mhz**
Hey all, 

This PR is to address this issue and milestone right here #628

Test were added, it's late here.. I didn't write tests for custom reducers. I can update this PR later with that.

### CODE REVIEW

**Comment by jmyrland** [comment]
Hi @s900mhz ! Thanks for looking into this.

With regards to the internal changes, @ctrlplusb has to do a review for this.

In the mean time, the API-docs & TS-types also needs to reflect these changes.

**Comment by s900mhz** [comment]
Sure thing! I'll work on the types and docs tonight.
I am open to any kind of discussion on the API. Just wanted to get this conversation started, we could really use this feature!

File: website/docs/docs/api/action.md [hunk 1]
```diff
@@ -48,6 +48,13 @@ An `action` is a function that is described below.
 
     The payload, if any, that was provided to the
     [action](/docs/api/action.html) when it was dispatched.
+  
+  - `config` (Object)
+
+    The config object that was provided to the
+    [action](/docs/api/action.html) when it was dispatched. 
+    - `immer` (Boolean)
+      Whether to use `immer` to update the state. Defaults to `true`.
```
Comment (ctrlplusb) [inline website/docs/docs/api/action.md:57]:
```suggestion
    - `immer` (Boolean)
      Whether to use `immer` to update the state. Defaults to `true`.
      
      You may want to consider disabling immer when dealing with a large/deep data structure within your state. Immer does unfortunately have an overhead as it wraps data in a Proxy solution for the mutation based updates to work. We would suggest only using this escape hatch if you are noticing any performance degradation.
```

File: website/docs/docs/api/reducer.md [hunk 2]
```diff
@@ -51,6 +51,12 @@ The `reducer` function is described below.
 
       Any payload that was provided to the action.
 
+  - `config` (Object)
+
+    The config object that was provided to the action when it was dispatched. 
+    - `immer` (Boolean)
+      Whether to use `immer` to update the state. Defaults to `true`.
```
Comment (ctrlplusb) [inline website/docs/docs/api/reducer.md:58]:
```suggestion
    - `immer` (Boolean)
      Whether to use `immer` to update the state. Defaults to `true`.
      
      You may want to consider disabling immer when dealing with a large/deep data structure within your state. Immer does unfortunately have an overhead as it wraps data in a Proxy solution for the mutation based updates to work. We would suggest only using this escape hatch if you are noticing any performance degradation.
```

**Comment by ctrlplusb** [comment]
Excellent stuff. Thanks for doing this. Just a few minors and we can get this merged. 🎉

**Comment by s900mhz** [comment]
@ctrlplusb  Awesome! I pushed the changes, lmk what you think!

**Review by jmyrland — APPROVED** [review]

**Comment by jmyrland** [comment]
Thanks for your contribution @s900mhz 👏 This will be included in the release this week!

**Pull request merged**